postgres/datadog_checks/postgres/statements.py

```diff
@@ -95,32 +91,47 @@ def _get_pg_stat_statements_columns(self, db):
         version is not a reliable way to determine the available columns on `pg_stat_statements`. The database can
         be upgraded without upgrading extensions, even when the extension is included by default.
         """
+        if len(self._stat_column_cache) == 0:
+            stat_column_query = PG_STAT_STATEMENTS_COLUMN_QUERY.format(
```

Comment:
It's not clear why this section was added and the purpose of this query. Can you explain the changes for getting the columns?

Reply:
@justiniso Yeah, most the explanation is [here](https://github.com/DataDog/integrations-core/pull/9365#discussion_r634566726). Just to add, basically we're running into an issue with new fields that are types that we do not want like strings and we need to filter them out. I've listed a few fields below that would cause issues if we didn't filter it out.

Types could include strings and NoneTypes which raises an error when we compute the derivative rows. This wasn't an issue before because we explicitly defined the columns we wanted in the [current integration](https://github.com/DataDog/integrations-core/blob/master/postgres/datadog_checks/postgres/statements.py#L69).
```
'rolbypassrls',
'rolcanlogin',
'rolpassword',
'rolsuper',
'rolvaliduntil',
'rolcreaterole',
```

Reply:
I had suggested caching the columns since there's no way they will change for the lifetime of the connection. We were redoing that 'get columns' query every time.

Reply:
Ah I see, ok that clarifies some things. I think this actually makes a stronger case for reverting to allowlisting the metric columns like we do now (https://github.com/DataDog/integrations-core/pull/9365#discussion_r634570939)

Reply:
@justiniso Yeah, I can agree with you on that. At the same time I also like how caching columns we want allows us to not have to maintain an allow list, but that doesn't require much maintenance. I'm definitely open to both suggestions 👍..

Reply:
I left a comment here https://github.com/DataDog/integrations-core/pull/9365#discussion_r634589257. I agree with the allowlist approach.

We can still cache the columns though. There's no reason to have to keep repeating that same query.

Reply:
Follow up, it looks like we're in favor of an allow list, so once I get confirmation on that, we'll go that route 🚀.